### ISSUE TITLE
Allow starting path with a named parameter

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/lucky_router#17]
Can add multiple paths with regex named params, but just one without regex, since only one per path length will work

So you can do:

* `/:organization`
* `/:organization/:repo` # Works since it is two parts so will not collide
* `/:username` # Will not work because it will never match! So raise
* `/:username[a-z]` # Will work! Router will first try to match against the regex and then fallback to the named param (`/:organization`). Now if in your app `:organization` and `:username` both wants a-z then you'll have an issue since the `/:username` will always match. In that case you need to do `/:organization_or_username` and figure it out in the action

Long story short, this will work, we just need to have some error handling so people don't get into weird situations where their paths won't ever match
